### PR TITLE
publish with delay

### PR DESCRIPTION
**Pull request by dhorelik**
publishing page with a delayed publish date

### CODE REVIEW

File: cms/tests/frontend/integration/publish.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+// #############################################################################
+// Users managed via the admin panel
```
Comment (vxsx) [inline cms/tests/frontend/integration/publish.js]: incorrect comment?

File: cms/tests/frontend/integration/publish.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            casper.echo(globals.baseUrl);
+            casper.echo(pageUrl);
+
+        })
+        // going to the newly created page url and checking that it hasn't been published yet
+        .thenOpen(pageUrl, function() {
+            test.assertTitle('Page not found', 'The page is not yet available');
+        })
+        // trying the same in a minute
+        .wait(60000)
```
Comment (vxsx) [inline cms/tests/frontend/integration/publish.js]: why?

File: cms/tests/frontend/integration/editContent.js [hunk 3]
```diff
@@ -52,15 +52,16 @@ casper.test.begin('Edit content', function (test) {
         .waitUntilVisible('.cms-modal-open')
         // change content inside appeared modal window
         .withFrame(0, function () {
-            casper.waitUntilVisible('#text_form', function () {
-                // explicitly put text to ckeditor
-                previousContentText = this.evaluate(function (contentData) {
-                    var previousContent = CMS.CKEditor.editor.document.getBody().getText();
+            casper
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/editContent.js]: I wouldn't add a new line here

File: cms/tests/frontend/integration/publish.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+
+casper.test.begin('Publishing dates', function (test) {
+    var pageUrl = (globals.baseUrl + 'Second').toLowerCase() + '/';
+    var pageTitle;
+    var publishDate;
+    var publishTime;
+
+    casper
+        .start(globals.editUrl)
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/publish.js]: as long as we are consistent with it it's also okay

File: cms/tests/frontend/integration/editContent.js (hunk 3 above)
Comment (vxsx) [inline cms/tests/frontend/integration/editContent.js]: yeah, wasn't me haha, will revert

File: cms/tests/frontend/integration/settings/messages.js [hunk 5]
```diff
@@ -75,6 +75,9 @@ module.exports = {
             deleteSuccessful: 'Delete plugin successful',
             cutSuccessful: 'Cut plugin successful',
             pasteAfterCutSuccessful: 'Paste after cutting plugin successful'
+        },
+        publish: {
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/settings/messages.js]: why is this empty?

Comment (vxsx) [inline cms/tests/frontend/integration/settings/messages.js]: it is supposed to have messages, although i don't know if it's really something we should keep maintaining

File: cms/tests/frontend/integration/publish.js (hunk 4 above)
Comment (vxsx) [inline cms/tests/frontend/integration/publish.js]: yeah, we are not really consistent with these :|

**Comment by FinalAngel** [comment]
LGTM :)

**Pull request merged**